### DISCUSSION TITLE
Editing a `Skin` resource

### DISCUSSION THREAD

**Discussion by juanjp600** [godotengine/godot-proposals#10850] [Editor]
I was looking at how skeletal animation is implemented for the purpose of extending [this library I'm making](https://github.com/juanjp600/duge.ToolLib.SceneBuilder/) to support skeletal animation.

As far as I can tell, this resource is only meant to be generated by an importer or otherwise by code, and never directly by a user. However, it is exposed in the editor like so:
![image](https://github.com/user-attachments/assets/8f3a637e-6d47-48c8-a0a1-f802b1a9decb)

Here are some issues that I found with this:
- Modifying the pose of a bind is not practical, as it's supposed to be [the inverse of the resting pose of the bone it's tied to.](https://github.com/godotengine/godot/blob/b2b13d46c2fcd75502eec12c768caa23ee952900/scene/3d/skeleton_3d.cpp#L881-L883)
- Changing the bind count does not immediately result in a refresh in the property editor. You're forced to deselect and reselect the resource to refresh the bind list. This is inconsistent with the observed behavior when modifying the name field of a bind, which does cause a refresh to change the visibility of the bone index field.
- The most important aspect of a skin IMO, which is the weights on each vertex, is a property of the mesh, not of the skin. Modifying the weights through the skin resource is not possible.

Given these limitations, I believe that the `Skin` resource should simply not expose any of its properties to the editor, which would prevent a user from misunderstanding it and making erroneous changes to it. This would be consistent with how [Skeleton3D does not directly expose all of its properties to the editor](https://github.com/godotengine/godot/blob/b2b13d46c2fcd75502eec12c768caa23ee952900/scene/3d/skeleton_3d.cpp#L187-L198), instead having specialized UI for this. However, I suspect that doing this might cause issues for a knowledgeable user. What are your thoughts?

**Comment by TokageItLab** [comment]
I agree that it would be impractical to manipulate it from the inspector, but I don't know if it needs to be hidden.

We need to be able to see what Skin it is and the inverse bind matrix for debugging purposes, so we need to at least show it even if we set it to readonly. However, it cannot be readonly because we need to allow changing it from code.

Given the above, I think it is okay to have an alternative GUI and replace it with that, but should not be approved to simply hide it.

**Reply by Calinou** [reply]
> However, it cannot be readonly because we need to allow changing it from code.

`PROPERTY_USAGE_READ_ONLY` only affects the inspector UI, not setting it from code.
